lib/PuppeteerSharp/Helpers/TaskHelper.cs

```diff
@@ -187,14 +192,32 @@ private static async Task<bool> TimeoutTask(Task task, TimeSpan timeout)
             }
 
             var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-            using (var cancellationToken = new CancellationTokenSource())
+            using (var cancellationToken = new CancellationTokenSource(timeout))
             {
-                cancellationToken.CancelAfter(timeout);
                 using (cancellationToken.Token.Register(s => ((TaskCompletionSource<bool>)s).TrySetResult(true), tcs))
                 {
                     return tcs.Task == await Task.WhenAny(task, tcs.Task).ConfigureAwait(false);
                 }
             }
         }
+
+        private static async Task<bool> TimeoutTask(Task task, TimeSpan timeout, CancellationToken cancellationToken)
+        {
+            if (timeout <= TimeSpan.Zero)
+            {
+                await task.ConfigureAwait(false);
+                return false;
+            }
+
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            using (var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+            {
+                linkedCancellationToken.CancelAfter(timeout);
+                using (linkedCancellationToken.Token.Register(s => ((TaskCompletionSource<bool>)s).TrySetResult(true), tcs))
+                {
+                    return tcs.Task == await Task.WhenAny(task, tcs.Task).ConfigureAwait(false);
+                }
+            }
```

Comment:
Can we use [using statements](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/using)?

Reply:
I've changed the outermost using statements to using declarations.

For the return value of `Register` I've kept the using statements, since
```cs
using var _ = linkedCancellationToken.Token.Register(s => ((TaskCompletionSource<bool>)s).TrySetResult(true), tcs);
```
triggers SA1312

See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3057#issuecomment-552490781